applications/tari_base_node/src/parser.rs

```diff
@@ -175,4 +178,82 @@ impl Parser {
             },
         }
     }
+
+    // Function to process  the get balance command
+    fn process_get_balance(&mut self) {
+        let mut handler = self.base_node_context.wallet_output_service.clone();
+        self.executor.spawn(async move {
+            let result = handler.get_balance().await;
+            if result.is_err() {
+                println!("Something went wrong");
+                warn!(
+                    target: LOG_TARGET,
+                    "Error communicating with wallet: {}",
+                    result.err().unwrap().to_string(),
+                );
+                return;
+            };
+            println!("Current balance is: {:?}", result.unwrap());
```

Comment:
`get_balance()` returns `Result<MicroTari,_>` right? It implements `Display`, so 
`println!("Current balance is: {}", result.unwrap());` should work

Reply:
No it returns: `Result<Balance, OutputManagerError>`
Which is good because balance is: 
```
pub struct Balance {
    /// The current balance that is available to spend
    pub available_balance: MicroTari,
    /// The current balance of funds that are due to be received but have not yet been confirmed
    pub pending_incoming_balance: MicroTari,
    /// The current balance of funds encumbered in pending outbound transactions that have not been confirmed
    pub pending_outgoing_balance: MicroTari,
}
```

I think its good te return all 3

Reply:
But I implemented display here.

Reply:
Ok, cool. The point was more to use `format!("{}")` vs `format!("{:?}")`